examples/enhanced_swarm_example.py

```diff
@@ -0,0 +1,190 @@
+
+    # Disable connection pool to maintain original behavior
+    connection_config = ConnectionConfig(enable_connection_pool=False)
+
+    # Create swarm with connection pool disabled
+    swarm = new_swarm(connection_config=connection_config)
+
+    logger.info("Backward compatibility mode:")
+    logger.info(
+        f"  Connection pool enabled: {connection_config.enable_connection_pool}"
+    )
+    logger.info(f"  Connections dict type: {type(swarm.connections)}")
+    logger.info("  Retry logic still available: 3 max retries")
+
+    await swarm.close()
+    logger.info("Backward compatibility example completed")
+
+
+async def example_production_ready_config() -> None:
+    """Example of production-ready configuration."""
+    logger.info("Creating enhanced Swarm with production-ready configuration...")
+
+    # Production-ready retry configuration
+    retry_config = RetryConfig(
+        max_retries=3,  # Reasonable retry limit
+        initial_delay=0.1,  # Quick initial retry
+        max_delay=30.0,  # Cap exponential backoff
+        backoff_multiplier=2.0,  # Standard exponential backoff
+        jitter_factor=0.1,  # Small jitter to prevent thundering herd
+    )
+
+    # Production-ready connection configuration
+    connection_config = ConnectionConfig(
+        max_connections_per_peer=3,  # Balance between performance and resource usage
+        connection_timeout=30.0,  # Reasonable timeout
+        enable_connection_pool=True,  # Enable for better performance
+        load_balancing_strategy="round_robin",  # Simple, predictable strategy
+    )
+
+    # Create swarm with production config
+    swarm = new_swarm(retry_config=retry_config, connection_config=connection_config)
+
+    logger.info("Production-ready configuration applied:")
+    logger.info(
+        f"  Retry: {retry_config.max_retries} retries, "
+        f"{retry_config.max_delay}s max delay"
+    )
+    logger.info(f"  Connections: {connection_config.max_connections_per_peer} per peer")
+    logger.info(f"  Load balancing: {connection_config.load_balancing_strategy}")
+
+    await swarm.close()
+    logger.info("Production-ready configuration example completed")
+
+
+async def main() -> None:
+    """Run all examples."""
+    logger.info("Enhanced Swarm Examples")
+    logger.info("=" * 50)
+
+    try:
+        await example_basic_enhanced_swarm()
+        logger.info("-" * 30)
+
+        await example_custom_retry_config()
+        logger.info("-" * 30)
+
+        await example_custom_connection_config()
+        logger.info("-" * 30)
+
+        await example_backward_compatibility()
+        logger.info("-" * 30)
+
+        await example_production_ready_config()
+        logger.info("-" * 30)
+
+        logger.info("All examples completed successfully!")
+
+    except Exception as e:
+        logger.error(f"Example failed: {e}")
+        raise
+
+
+if __name__ == "__main__":
+    asyncio.run(main())
```

Comment:
Would be nice if you could move this swarm example in examples/doc-examples or another directory and update the documentation (docs/ rst )